doc/nom-guide/src/chapter_2.md

```diff
@@ -0,0 +1,105 @@
+# Chapter 2: Tags and Character Classes
+
+The simplest _useful_ regex you can write is one which
+has no special characters, it just matches a string.
+
+Imagine, for example, the regex `/abc/`. It simply matches when the string
+`"abc"` occurs.
+
+In `nom`, we call a simple collection of bytes a tag. Because
+these are so common, there already exists a function called `tag()`.
+This function returns a parser for a given string.
+
+<div class="example-wrap" style="display:inline-block"><pre class="compile_fail" style="white-space:normal;font:inherit;">
+ **Warning**: `nom` has multiple different definitions of `tag`, make sure you use this one for the
+ moment!
+</pre></div>
+
+```rust
+# extern crate nom;
+pub use nom::bytes::complete::tag;
+```
+
+For example, the regex `/abc/` (really, the regex `/^abc/`)
+could be represented as `tag("abc")`.
+
+Note, that the function `tag` will return
+another function, namely, a parser for the tag you requested.
+
+Below, we see a function using this:
+
+```rust
+# extern crate nom;
+# pub use nom::bytes::complete::tag;
+# pub use nom::IResult;
+
+fn parse_input(input: &str) -> IResult<&str, &str> {
+    //  note that this is really creating a function, the parser for abc
+    //  vvvvv 
+    //         which is then called here, returning an IResult<&str, &str>
+    //         vvvvv
+    tag("abc")(input)
+}
+
+    let ok_input = "abcWorld";
+
+    match parse_input(ok_input) {
+        Ok((leftover_input, output)) => {
+            assert_eq!(leftover_input, "World");
+            assert_eq!(output, "abc");
+        },
+        Err(_) => unreachable!()
+    }
+
+    let err_input = "defWorld";
+    match parse_input(err_input) {
+        Ok((leftover_input, output)) => unreachable!(),
```

Comment:
if all those examples are only looking at one branch, they should use `unwrap` and `unwrap_err` instead of a `match` with one branch using `unreachable!()`, that will make them shorter

Reply:
I've chosen to go with returning a `Box<dyn Error>`, so that I can use Try. I've done this because (from my limited understanding) the Rust guide to writing documentation suggests avoiding `unwrap` in documentation examples. 

Happy to change this if you think it's the wrong direction.